src/annotator/components/NotebookModal.js

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useRef, useState } from 'preact/hooks';
+import classnames from 'classnames';
+
+import { createSidebarConfig } from '../config/sidebar';
+
+// FIXME: use the button from the frontend shared package once this is stable.
+import Button from '../../sidebar/components/Button';
+
+/**
+ *
+ * @typedef NotebookIframeProps
+ * @prop {Record<string, any>} config
+ * @prop {string} groupId
+ */
+
+/**
+ * Create the iframe that will load the notebook application.
+ * 
+ * @param {NotebookIframeProps} props
+
+ */
+function NotebookIframe({ config, groupId }) {
+  const notebookConfig = createSidebarConfig(config);
+  // Explicity set the "focused" group
+  notebookConfig.group = groupId;
+  const configParam = encodeURIComponent(JSON.stringify(notebookConfig));
+  const notebookAppSrc = `${config.notebookAppUrl}#config=${configParam}`;
+
+  return (
+    <iframe
+      key={groupId} // force re-rendering on group change
+      title={'Hypothesis annotation notebook'}
+      className="Notebook__iframe"
+      // Enable media in annotations to be shown fullscreen
+      allowFullScreen
+      src={notebookAppSrc}
+    />
+  );
+}
+
+/**
+ * @typedef NotebookModalProps
+ * @prop {import('../util/emitter').EventBus} eventBus
+ * @prop {Record<string, any>} config
+ */
+
+/**
+ *
+ * @param {NotebookModalProps} props
+ */
+export default function NotebookModal({ eventBus, config }) {
+  const [isHidden, setIsHidden] = useState(true);
+  const [groupId, setGroupId] = useState(/** @type {string|null} */ (null));
+  const originalDocumentOverflowStyle = useRef('');
+  const emitter = useRef(
+    /** @type{ReturnType<eventBus['createEmitter']>|null} */ (null)
+  );
+
+  useEffect(() => {
+    originalDocumentOverflowStyle.current = document.body.style.overflow;
```

Comment:
I think it might help readability here to split all the document-style related logic into a separate layout effect, which depends on `isHidden`. This would allow grouping all such logic together.

Reply:
I was short of thinking about that. Thanks for the suggestion. I implemented it.